### PR TITLE
👏  making request and emit currying optional

### PR DESCRIPTION
**Pull request by jacktuck**
- [x] Implement currying
- [x] Add tests for currying
~- [ ] Write documentation~ will put docs in a new PR

### CODE REVIEW

**Comment by jacktuck** [comment]
WIP.

Remind me to add some docs for this.

File: test/emitter.test.js [hunk 1]
```diff
@@ -18,6 +18,49 @@ describe('Emitter', function () {
...
+            expect(event.data).to.equal('had a jalfrezi')
+            done()
+          } catch (err) {
+            done(err)
+          }
+
+          return event.data
+        })
+        .start()
+        .then(_ => remit.emit('had_jalfrezi')('had a jalfrezi'))
```
Comment (jpwilliams) [inline test/emitter.test.js]: Need to check that what comes back here is a promise that resolves once the emission has been sent as opposed to just an Emission object.

**Comment by jpwilliams** [comment]
This was looking all good, by the way. Tests needs to change very slightly to check that the correct types are being concerned.

I have some stashed that I started making for this if you don't have time! I'll look for them.

File: test/emitter.test.js (hunk 1 above)
Comment (jacktuck) [reply test/emitter.test.js]: How would i be able to go about that? Not sure theres any way to tell if emission is received or not. Could log Date.now in the endpoint and after resolution, is that enough, is it flimsy?

Comment (jpwilliams) [reply test/emitter.test.js]: Ah no it’s literally just checking that `remit.emit` returns a _promise_ (for the emission being sent) or an _Emission_ object when appropriate, to check that the currying it doing the correct thing. :)

No need to worry about if the emission was received or not.

Comment (jacktuck) [reply test/emitter.test.js]: Oh! Alright i'll take a look!

**Comment by jpwilliams** [comment]
Looking back on this, I worry that it acts as a sort of leak.

If I make a request via `const result = await remit.request('foo', 'myData')`, I wouldn't assume that the request is still active and waiting to be reused, as I can't actually get access to it.

Either this needs dropping or some more work added to "close" the consumer after sending the single request. We should probably discourage that, though, as the cost of creating/deleting lots of request channels gets pricey.

🤔

**Comment by jpwilliams** [comment]
Closing this as there's nothing being continued here, but I'll keep the branch around if we ever get around to adding a `close()` method to requests.

**Pull request closed without merging**